AEPIdentity/Sources/IdentityProperties.swift

```diff
@@ -51,7 +51,9 @@ struct IdentityProperties: Codable {
     func toEventData() -> [String: Any] {
         var eventData = [String: Any]()
         eventData[IdentityConstants.EventDataKeys.VISITOR_ID_ECID] = ecid?.ecidString
-        eventData[IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER] = advertisingIdentifier
+        if let adId = advertisingIdentifier, !adId.isEmpty {
```

Comment:
We should add this same `isEmpty` check for pushIdentifier, blob, and locationHint.